src/WpfMath/TexFormulaHelper.cs

```diff
@@ -153,9 +153,34 @@ public void AddOperator(string operatorFormula, string lowerLimitFormula, string
                 formulaParser.Parse(upperLimitFormula), useVerticalLimits);
         }
 
+        public int MoveOffset(Atom atom, int pos)
```

Comment:
This method hasn't the best name (probably `RecursivelyChangeSourceOffset` would sound better?). Also, should it be `private`?

We definitely need a comment with a motivation (why do we ever need to change a source offset? Because of the predefined formulae, of course).

Also, I think it could be marked as `internal` and tested separately? Would it make sense?